packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/ThrowStatement.cs

```diff
@@ -5,8 +5,15 @@
 
 namespace Microsoft.Generator.CSharp.Statements
 {
-    public sealed record ThrowStatement(ValueExpression ThrowExpression) : MethodBodyStatement
+    public sealed class ThrowStatement : MethodBodyStatement
     {
+        public ValueExpression ThrowExpression { get; }
+
+        public ThrowStatement(ValueExpression throwExpression)
+        {
+            ThrowExpression = throwExpression;
+        }
```

Comment:
so this is like a wrapper of the `throw expression` to convert it to a statement.

If this is the case, it does not make sense that the type of this expression is just a `ValueExpression`.
What if they do this:
```
var expression = new AssignmentExpression();
var statement = new ThrowStatement(expression); // this will write a thing like `i = 1;` which is not a throw expression at all.
```

I think we should either let it have the same structure as `ThrowExpression`, or let this encapsulate a `ThrowExpression`

Also maybe we need to add an implicit operator from throwExpression as well?

Reply:
We don't have a ThrowExpression.  Since this is not introduced with my PR can you open / link a follow up item if we think we should tweak this?